mypy/main.py

```diff
@@ -46,7 +47,9 @@ def stat_proxy(path: str) -> os.stat_result:
         return st
 
 
-def main(script_path: Optional[str], args: Optional[List[str]] = None) -> None:
+def main(script_path: Optional[str], args: Optional[List[str]] = None,
+         stdout: TextIO = sys.stdout,
+         stderr: TextIO = sys.stderr) -> None:
```

Comment:
Again... (Maybe this is the toplevel where you should check for `None`?)

Reply:
This is not quite the top level, because a fair few things call `build` directly (mostly tests but adding this parameter they don't care about just so they keep working exactly as they currently do seems dumb). I tried doing the `None` checking here but mypy couldn't infer that `f` in `flush_errors` couldn't be `None` after that, so I'm having to pass in actual values in `__main__.py` and instead do the checking in `build`